AstSemantics.md

```diff
@@ -273,39 +279,53 @@ nested. This guarantees that all resulting control flow graphs are well-structur
 
 ## Calls
 
-Direct calls to a function specify the callee by index into a function table.
+Each function has a *signature*, which consists of:
 
-  * `call_direct`: call function directly
+  * Return types, which are a sequence of local types
+  * Argument types, which are a sequence of local types
 
-Each function has a signature in terms of expression types, and calls must match
-the function signature
-exactly. [Imported functions](MVP.md#code-loading-and-imports) also have
-signatures and are added to the same function table and are thus also callable
-via `call_direct`.
+Note that WebAssembly itself does not support variable-length argument lists
+(aka varargs). C and C++ compilers are expected to implement this functionality
+by storing arguments in a buffer in linear memory and passing a pointer to the
+buffer.
 
-Indirect calls may be made to a value of function-pointer type. A function-
-pointer value may be obtained for a given function as specified by its index
-in the function table.
+In the MVP, the length of the return types vector may only be 0 or 1. This
+restriction may be lifted in the future with the addition of support for
+[multiple return values](FutureFeatures.md#multiple-return-values").
```

Comment:
I don't think the PR represents consensus yet, so maybe it's more
appropriate to make an issue for discussion first?

On Thu, Jul 23, 2015 at 11:02 PM, Dan Gohman <notifications@github.com>
wrote:

> In AstSemantics.md
> <https://github.com/WebAssembly/design/pull/278#discussion_r35370932>:
>
> >
> > -Indirect calls may be made to a value of function-pointer type. A function-
> > -pointer value may be obtained for a given function as specified by its index
> > -in the function table.
> > +In the MVP, the length of the return types vector may only be 0 or 1. This
> > +restriction may be lifted in the future with the addition of support for
> > +[multiple return values](FutureFeatures.md#multiple-return-values").
>
> Yep, fixed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/design/pull/278/files#r35370932>.
>